### PR TITLE
Fix building KuzzleErrors with non KuzzleError for errorsManager

### PR DESCRIPTION
**Pull request by Yoann-Abbes**
## What does this PR do ?

This PR fixes the cases where a non KuzzleError is handled by the errorsManager.
Before it, the stack trace was override.
Now it properly builds the error and retrieves the stack-trace and the message

### How should this be manually tested?

When you define an error, you just have to name it starting with `derived` so KuzzleError class will handle it in its constructor.

### CODE REVIEW

File: lib/api/core/entrypoints/protocols/http.js [hunk 1]
```diff
@@ -191,6 +191,9 @@ class HttpProtocol extends Protocol {
 
     // We forwarded all pipe errors to the request's event handler
     request.on('error', err => {
+      const kerr = err instanceof KuzzleError
```
Comment (kuzzle) [inline lib/api/core/entrypoints/protocols/http.js]: MAJOR Remove this useless assignment to local variable "kerr" [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1854)
MINOR Remove the declaration of the unused 'kerr' variable. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AUnusedVariable)

File: lib/api/core/entrypoints/protocols/http.js [hunk 2]
```diff
@@ -227,7 +230,11 @@ class HttpProtocol extends Protocol {
 
       this._compress(data, response, message.headers, (err, deflated) => {
         if (err) {
-          this._replyWithError(message, response, err);
+          const kuzerr = err instanceof KuzzleError 
+            ? err
+            : errorsManager.getErrorFrom(err, 'http_request_error');
+
+          this._replyWithError(connection, proxyRequest, response, kuzerr);
```
Comment (kuzzle) [inline lib/api/core/entrypoints/protocols/http.js]: CRITICAL "_replyWithError" expects 3 arguments, but 4 were provided. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS930)

**Comment by kuzzle** [comment]
SonarQube analysis reported 4 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L968](https://github.com/kuzzleio/kuzzle/blob/1effbcea033e7b78d0a685c5a548161d0ee2ffc3/lib/api/core/plugins/pluginsManager.js#L968): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L968](https://github.com/kuzzleio/kuzzle/blob/2e2bace1ac2490b72939dc187cd2844c3352fdb2/lib/api/core/plugins/pluginsManager.js#L968): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: lib/config/error-codes/throw.js [hunk 3]
```diff
@@ -67,39 +68,63 @@ function throwError (domain, subdomain, error, ...placeholders) {
...
+ 
+ * @param  {string} errorSource - non KuzzleError from which the KuzzleError will derivate
  * @param  {string} domain - Domain (eg: 'external')
  * @param  {string} subdomain - Subdomain (eg: 'elasticsearch')
  */
 function wrap (domain, subdomain) {
   return {
     getError: (error, ...placeholders) => getError(domain, subdomain, error, ...placeholders),
     throw: (error, ...placeholders) => throwError(domain, subdomain, error, ...placeholders),
-    reject: (error, ...placeholders) => rejectError(domain, subdomain, error, ...placeholders)
```
Comment (Aschen) [inline lib/config/error-codes/throw.js]: Why removing the `reject` method?

Comment (Yoann-Abbes) [reply lib/config/error-codes/throw.js]: hmm I assume its a mistake when I rebased.

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L968](https://github.com/kuzzleio/kuzzle/blob/52d9bb6f63ae57c60295b8b9bf896e04b7b9b5ed/lib/api/core/plugins/pluginsManager.js#L968): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: lib/config/error-codes/throw.js [hunk 4]
```diff
@@ -75,31 +75,69 @@ function throwError (domain, subdomain, error, ...placeholders) {
  * @param  {string} error - Error name: (eg: 'index_not_found')
  * @param  {...any} placeholders - Placeholders value to inject in error message
  */
-function rejectError (domain, subdomain, error, ...placeholders) {
+
+function rejectError(domain, subdomain, error, ...placeholders) {
```
Comment (Aschen) [inline lib/config/error-codes/throw.js]: The rejection should use `Bluebird` instead of `Promise`

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L968](https://github.com/kuzzleio/kuzzle/blob/6d9dc3bd16fcc94af59158ece442ff2e25d85e20/lib/api/core/plugins/pluginsManager.js#L968): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L968](https://github.com/kuzzleio/kuzzle/blob/ee9336f3e356e991984711531aa6909870b25e87/lib/api/core/plugins/pluginsManager.js#L968): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L968](https://github.com/kuzzleio/kuzzle/blob/f063d900421256da86ec228d8c259bdbea5908a4/lib/api/core/plugins/pluginsManager.js#L968): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L968](https://github.com/kuzzleio/kuzzle/blob/6a601d0c4ab506ac3ae670c3032dbc20950787fb/lib/api/core/plugins/pluginsManager.js#L968): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L968](https://github.com/kuzzleio/kuzzle/blob/459883faa8fb066bca76a8b8dd9ce144429c3cd5/lib/api/core/plugins/pluginsManager.js#L968): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L968](https://github.com/kuzzleio/kuzzle/blob/f048ed88bf30c661b922eeb8f8d46cd2dbd1b782/lib/api/core/plugins/pluginsManager.js#L968): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L968](https://github.com/kuzzleio/kuzzle/blob/271c67edca0c0c76de104c1b11cf36a3c2878503/lib/api/core/plugins/pluginsManager.js#L968): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Pull request merged**